pageserver/src/pgdatadir_mapping.rs

```diff
@@ -524,7 +524,9 @@ impl Timeline {
                 total_size += relsize as u64;
             }
         }
-        Ok(total_size * BLCKSZ as u64)
+        let calculated_size = total_size * BLCKSZ as u64;
+        self.metrics.current_logical_size_gauge.set(calculated_size);
```

Comment:
This is the wrong place to set the gauge, because this is "calculate non-incremental logical size at any lsn", which is just incidentially called by "calculate initial logical size". It is used by tenant_size (synthetic_size inputs gathering).